### PR TITLE
Restrict metadata fuzzy scan filter using start and stop keys

### PR DESCRIPTION
**Pull request by poornachandra**
Build - http://builds.cask.co/browse/CDAP-DUT3843-1

### CODE REVIEW

**Comment by sagarkapare** [comment]
LGTM :+1:

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java [hunk 1]
```diff
@@ -451,7 +455,9 @@ private Metadata getSnapshotBeforeTime(Id.NamespacedId targetId, long timeMillis
     // Scan using fuzzy filter. Scan returns one row per property.
     // Group the rows on entityId
     Multimap<Id.NamespacedId, MetadataEntry> metadataMap = HashMultimap.create();
-    Scanner scan = indexedTable.scan(new Scan(null, null, new FuzzyRowFilter(fuzzyKeys)));
+    byte[] start = fuzzyKeys.get(0).getFirst();
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java]: this method can't be called with empty set of `targetIds`?

Comment (poornachandra) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java]: We check for empty `targetIds` in lines 443-445 above.

**Comment by poornachandra** [comment]
Build is green, merging.

**Pull request merged**